### ISSUE TITLE
get_text("words") for only one page

### ISSUE THREAD

**Issue by DariaIzzo** [pymupdf/PyMuPDF#1536]
Hello

I am using pymupdf version 1.19.4
I would like to use the function get_text("words") in order to extract a single page in a multi page pdf.

My code is:

import fitz
doc = fitz.open("multi_page_file.pdf)
out_pymupdf = doc[0].get_text("words")
pdf_width, pdf_height = list(page.MediaBox)[2:]

for idx in range(len(out_pymupdf)):
     if not (out_pymupdf[idx][2] <= pdf_width and out_pymupdf[idx][3] <= pdf_height):
            print(out_pymupdf[idx])

This code shows that in out_pymupdf we also have the output from the second page of the pdf, where the bounding boxes exceed the width and height of the page. However, some words on the second page still have smaller coordinates as the pdf_height (which is strange as well).

If I do all this using doc[1], I again get the output of the entire PDF but I get negative numbers for the y-axis coordinates for words of the first page.

If I use doc[0].get_text() I only get the text from the first page. So the functions get_text() and get_text("words") are not consistent.

How can I get only the requested page from the function get_text("words"), such that all bounding box coordinates are within 0-width/height of the page?

**Comment by JorjMcKie** [comment]
I don't understand your explanation.
For a **_bug_** the mandatory reproducer is missing.

**Comment by DariaIzzo** [comment]
Take any PDF that contains more that 1 page and apply this code:

doc = fitz.open("multi_page_file.pdf)
out_pymupdf = doc[0].get_text("words")

I expect out_pymupdf to have only the output for the first page in the PDF as I apply the fuction to doc[0]. However, I get the output for the entire PDF. This is an unexpected behaviour so a bug? Correct me if I am wrong.

Thanks
Daria

**Comment by DariaIzzo** [comment]
Thank you for your answers!
Indeed, here the output is right.
I tought that it is a general problem as it was false for all my PDFs. In that case my PDFs are in some kind special.
I will try to reproduce the error as I cannot give you my original data.

**Comment by JorjMcKie** [comment]
You can send a file to my private e-mail any time. I would keep it confidential.
For the time being I am going to close this issue.
Reopen it when you have all required information.

As an aside:
That method extracts words from a so-called TextPage object, which is a wrapper for a corresponding MuPDF object.
So your obervations would imply that **_MuPDF_** extracts all document pages to one TextPage (which I then extract) - and all that just because `doc[0]` is used instead of a page object defined as `page = doc[0]` -- of which MuPDF does not know a hack about?

All this is so far from being believable that I am eager to see one example exhbiting this miracle.
